wpilibj/src/main/java/edu/wpi/first/wpilibj/SharpIR.java

```diff
@@ -0,0 +1,146 @@
+// Copyright (c) FIRST and other WPILib contributors.
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package edu.wpi.first.wpilibj;
+
+import edu.wpi.first.hal.SimDevice;
+import edu.wpi.first.hal.SimDevice.Direction;
+import edu.wpi.first.hal.SimDouble;
+import edu.wpi.first.util.sendable.Sendable;
+import edu.wpi.first.util.sendable.SendableBuilder;
+import edu.wpi.first.util.sendable.SendableRegistry;
+
+/**
+ * SharpIR analog distance sensor class. These distance measuring sensors output an analog voltage
+ * corresponding to the detection distance.
+ *
+ * <p>Teams are advised that the case of these sensors are conductive and grounded, so they should
+ * not be mounted on a metallic surface on an FRC robot.
+ */
+@SuppressWarnings("MethodName")
+public class SharpIR implements Sendable, AutoCloseable {
+  private AnalogInput m_sensor;
+
+  private SimDevice m_simDevice;
+  private SimDouble m_simRange;
+
+  private final double m_A;
+  private final double m_B;
+  private final double m_minCM;
+  private final double m_maxCM;
+
+  /**
+   * Sharp GP2Y0A02YK0F is an analog IR sensor capable of measuring distances from 20cm to 150cm.
+   *
+   * @param channel Analog input channel the sensor is connected to
+   * @return sensor object
+   */
+  public static SharpIR GP2Y0A02YK0F(int channel) {
+    return new SharpIR(channel, 62.28, -1.092, 20, 150.0);
+  }
+
+  /**
+   * Sharp GP2Y0A21YK0F is an analog IR sensor capable of measuring distances from 10cm to 80cm.
+   *
+   * @param channel Analog input channel the sensor is connected to
+   * @return sensor object
+   */
+  public static SharpIR GP2Y0A21YK0F(int channel) {
+    return new SharpIR(channel, 26.449, -1.226, 10.0, 80.0);
+  }
+
+  /**
+   * Sharp GP2Y0A41SK0F is an analog IR sensor capable of measuring distances from 4cm to 40cm.
+   *
+   * @param channel Analog input channel the sensor is connected to
+   * @return sensor object
+   */
+  public static SharpIR GP2Y0A41SK0F(int channel) {
+    return new SharpIR(channel, 12.84, -0.9824, 4.0, 40.0);
```

Comment:
my team used 12.354 and -1.07

Reply:
I don't know where I got my constants from but yours seem fine too.